rudder-web/src/main/scala/com/normation/rudder/web/rest/compliance/ComplianceAPIData.scala

```diff
@@ -0,0 +1,248 @@
+import net.liftweb.json.JsonDSL._
+import com.normation.rudder.domain.policies.DirectiveId
+
+
+
+/**
+ * Here, we want to present two view of compliance:
+ * - by node
+ * - by rule
+ *
+ * For "by node", we want to have a list of node, then rules,
+ * then directives, then component and then values.
+ *
+ * For "by rule", we want to have a list of rules, then
+ * directives, then components, THEN NODEs, then values.
+ * Because we want to be able to analyse at the component level
+ * if we have problems on several nodes, but values are only
+ * meaningful in the context of a node.
+ *
+ * So we define two hierarchy of classes, with the name convention
+ * starting by "ByRule" or "ByNode".
+ */
+
+
+/**
+ * Compliance for a rules.
+ * It lists:
+ * - id: the rule id
+ * - compliance: the compliance of the rule by node
+ *   (total number of node, repartition of node by status)
+ * - nodeCompliance: the list of compliance for each node, for that that rule
+ */
+case class ByRuleRuleCompliance(
+    id             : RuleId
+    //compliance by nodes
+  , compliance     : ComplianceLevel
+  , directives: Seq[ByRuleDirectiveCompliance]
+)
+
+case class ByRuleDirectiveCompliance(
+    id        : DirectiveId
+  , compliance: ComplianceLevel
+  , components: Seq[ByRuleComponentCompliance]
+)
+
+case class ByRuleComponentCompliance(
+    name      : String
+  , compliance: ComplianceLevel
+  , nodes     : Seq[ByRuleNodeCompliance]
+)
+
+case class ByRuleNodeCompliance(
+    id        : NodeId
+  , values    : Seq[ComponentValueStatusReport]
+)
+
+/**
+ * Compliance for a node.
+ * It lists:
+ * - id: the node id
+ * - compliance: the compliance of the node by rule
+ *   (total number of rules, repartition of rules by status)
+ * - ruleUnderNodeCompliance: the list of compliance for each rule, for that node
+ */
+case class ByNodeNodeCompliance(
+    id             : NodeId
+    //compliance by nodes
+  , compliance     : ComplianceLevel
+  , nodeCompliances: Seq[ByNodeRuleCompliance]
+)
+
+/**
+ * Compliance for rule, for a given set of directives
+ * (normally, all the directive for a given rule)
+ * It lists:
+ * - id: the rule id
+ * - compliance: total number of directives and
+ *   repartition of directive compliance by status
+ * - directiveCompliances: status for each directive, for that rule.
+ */
+case class ByNodeRuleCompliance(
+    id        : RuleId
+    //compliance by directive (by nodes)
+  , compliance: ComplianceLevel
+  , directives: Seq[ByNodeDirectiveCompliance]
+)
+
+case class ByNodeDirectiveCompliance(
+    id        : DirectiveId
+  , compliance: ComplianceLevel
+  , components: Map[String, ComponentStatusReport]
+)
+
+object ByNodeDirectiveCompliance {
+
+  def apply(d: DirectiveStatusReport): ByNodeDirectiveCompliance = {
+    new ByNodeDirectiveCompliance(d.directiveId, d.compliance, d.components)
+  }
+}
+
+object JsonCompliance {
+
+
+  implicit class JsonbyRuleCompliance(rule: ByRuleRuleCompliance) {
```

Comment:
I'm not sur, because in the UI, you have the two view availables, so having nodes and value merge may not be a limitation. 
On the other hand, we could have a way to see which nodes are in error for a given value (and only errors), what could be nice